src/persist-client/src/write.rs

```diff
@@ -461,13 +461,13 @@ where
                     handle_shard: self.machine.shard_id(),
                 });
             }
-            if self.cfg.build_version != batch.version {
-                error!(
+            check_data_version(&self.cfg.build_version, &batch.version);
+            if self.cfg.build_version > batch.version {
+                info!(
```

Comment:
Yeah, I'd consider a `// TODO: error on very old versions once the leaked blob detector exists', possibly as part of the error message. But up to you!

Reply:
Done! TFTR!